compiler/next/include/chpl/types/UnknownType.h

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2021 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CHPL_TYPES_UNKNOWNTYPE_H
+#define CHPL_TYPES_UNKNOWNTYPE_H
+
+#include "chpl/types/Type.h"
+
+namespace chpl {
+namespace types {
+
+
+/**
+  This class represents an unknown type which can be used to represent a case
+  where the type is not established because it is generic.
+ */
+class UnknownType : public Type {
+ private:
+  UnknownType() : Type(typetags::UnknownType) { }
+
+  bool contentsMatchInner(const Type* other) const override {
+    return true;
+  }
+
+  void markUniqueStringsInner(Context* context) const override {
+  }
+
+  bool isGeneric() override {
+    return false;
```

Comment:
Should this return true?  It seems like it might cause problems for it not to, given your comment on the class.

Reply:
I'll make it return true for now, but of course it might change again once we get to using it.